packages/core/core/src/services/entity-validator/validators.ts

```diff
@@ -138,7 +139,13 @@ const addStringRegexValidator = (
  */
 const addUniqueValidator = <T extends strapiUtils.yup.AnySchema>(
   validator: T,
-  { attr, model, updatedAttribute, entity }: ValidatorMetas<Attribute.Any & Attribute.UniqueOption>
+  {
+    attr,
+    model,
+    updatedAttribute,
+    entity,
+    locale,
+  }: ValidatorMetas<Attribute.Any & Attribute.UniqueOption>
```

Comment:
I am in the process of manually testing this, but I believe we forgot about the publication state of the document.

Fields should be unique across the same locale, but also across the same publication state.